src/main/java/io/github/thebusybiscuit/slimefun4/core/guide/unlockprovider/CurrencyUnlockProvider.java

```diff
@@ -0,0 +1,37 @@
+package io.github.thebusybiscuit.slimefun4.core.guide.unlockprovider;
+
+import javax.annotation.Nonnull;
+
+import org.bukkit.entity.Player;
+
+import io.github.thebusybiscuit.slimefun4.api.researches.Research;
+import io.github.thebusybiscuit.slimefun4.api.guide.SlimefunGuideUnlockProvider;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.integrations.VaultIntegration;
+
+/**
+ * Unlock research by withdrawing player's balance.
+ *
+ * @see SlimefunGuideUnlockProvider
+ */
+public final class CurrencyUnlockProvider implements SlimefunGuideUnlockProvider {
```

Comment:
Is there a reason these are final?
It only prevents someone else from extending it which I feel like we _should_ allow